internal/querynodev2/delegator/delegator.go

```diff
@@ -375,12 +381,7 @@ func (sd *shardDelegator) Search(ctx context.Context, req *querypb.SearchRequest
 			}
 			results[i] = result
 		}
-		var ret *internalpb.SearchResults
```

Comment:
I removed useless reduce here, and return directly for reduction outside in searchChannel